pkg/network/tracer/cilium_lb.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	backends, err := ebpf.LoadPinnedMap("/sys/fs/bpf/tc/globals/cilium_lb4_backends_v3", &ebpf.LoadPinOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error loading pinned backends map: %w", err)
+	}
+
+	clb := &ciliumLoadBalancerConntracker{
+		backendIDToBackend: make(map[uint32]backend),
+	}
+	if clb.ctTCP, err = maps.Map[tupleKey4, ctEntry](ctTCP); err != nil {
+		return nil, fmt.Errorf("could not make generic map for ct TCP map: %w", err)
+	}
+	if clb.ctUDP, err = maps.Map[tupleKey4, ctEntry](ctUDP); err != nil {
+		return nil, fmt.Errorf("could not make generic map for ct UDP map: %w", err)
+	}
+	if clb.backends, err = maps.Map[backend4KeyV3, backend4ValueV3](backends); err != nil {
+		return nil, fmt.Errorf("could not make generic map for backends map: %w", err)
+	}
+
+	clb.stop = make(chan struct{})
+	go func() {
+		tick := time.NewTicker(10 * time.Second)
+		defer tick.Stop()
+		for {
+			select {
+			case <-tick.C:
+				clb.updateBackends()
+			case <-clb.stop:
+				close(clb.stop)
+				return
+			}
+		}
+	}()
+
+	return clb, nil
+}
+
+func ntohs(n uint16) uint16 {
+	return binary.BigEndian.Uint16([]byte{byte(n), byte(n >> 8)})
+}
+
+func htons(n uint16) uint16 {
+	b := make([]byte, 2)
+	binary.BigEndian.PutUint16(b, n)
+	return *(*uint16)(unsafe.Pointer(&b[0]))
+}
+
+func (clb *ciliumLoadBalancerConntracker) updateBackends() {
+	clb.m.Lock()
+	defer clb.m.Unlock()
+
+	it := clb.backends.Iterate()
+	var k backend4KeyV3
+	var v backend4ValueV3
+	for it.Next(&k, &v) {
+		clb.backendIDToBackend[k.ID] = backend{
+			addr: util.AddressFromNetIP(net.IPv4(v.Address[0], v.Address[1], v.Address[2], v.Address[3])),
+			port: ntohs(v.Port),
+		}
+	}
+
+	if it.Err() != nil {
+		log.Warnf("error iterating lb backends map: %s", it.Err())
+	}
+}
+
+func (clb *ciliumLoadBalancerConntracker) Describe(chan<- *prometheus.Desc) {
+}
+
+// Collect returns the current state of all metrics of the collector
+func (clb *ciliumLoadBalancerConntracker) Collect(chan<- prometheus.Metric) {
+}
+
+func (clb *ciliumLoadBalancerConntracker) GetTranslationForConn(c *network.ConnectionTuple) *network.IPTranslation {
+	// TODO: add ipv6 support
+	if c.Family != network.AFINET {
+		return nil
+	}
+
+	if c.Direction != network.OUTGOING {
+		return nil
+	}
+
+	if c.Dest.IsLoopback() {
+		return nil
+	}
+
+	queryMap := clb.ctTCP
+	t := tupleKey4{
+		Flags:      TUPLE_F_OUT | TUPLE_F_SERVICE,
+		NextHeader: uint8(unix.IPPROTO_TCP),
+		SourcePort: htons(c.SPort),
+		DestPort:   htons(c.DPort),
+		SourceAddr: c.Source.As4(),
+		DestAddr:   c.Dest.As4(),
+	}
+	if c.Type == network.UDP {
+		t.NextHeader = unix.IPPROTO_UDP
+		queryMap = clb.ctUDP
+	}
+
+	log.TraceFunc(func() string {
+		return fmt.Sprintf("looking up tuple %+v in ct map", t)
+	})
+
+	var ctEntry ctEntry
+	var err error
+	if err = queryMap.Lookup(&t, &ctEntry); err != nil {
+		if !errors.Is(err, ebpf.ErrKeyNotExist) {
+			log.Warnf("error looking up %+v in ct map: %s", t, err)
```

Comment:
should we add any prometheus telemetry around these errors?

Reply:
Added some telemetry in [`c2b5824` (#31842)](https://github.com/DataDog/datadog-agent/pull/31842/commits/c2b5824799284b60f780f9a6ea9827f88242a0e5). We don't really track errors like this in any of the other conntrackers. Without knowing what error, this would not be very useful.

Reply:
Could we have enough errors for this log to flood?